src/components/Hits.js

```diff
@@ -1,10 +1,18 @@
 import React from 'react';
-import { connectHits, Pagination } from 'react-instantsearch-dom';
+import { connectHits } from 'react-instantsearch-dom';
 
 import { useAppContext } from '../hooks';
+import { Pagination } from './Pagination';
 
 export const Hits = connectHits((props) => {
-  const { view, ConnectedHit } = useAppContext();
+  const { view, ConnectedHit, topAnchor } = useAppContext();
+
+  const onPageClick = React.useCallback(
```

Comment:
We have to memoize functions passed to React InstantSearch components (`Pagination` here) otherwise it triggers an infinite loop.